pkg/runner/local_exec.go

```diff
@@ -68,7 +71,7 @@ func (r *LocalExecutableRunner) Run(ctx context.Context, input *api.RunInput, ow
 			logging.S().Info("temporary redis instance started successfully")
 		} else {
 			close(redisWaitCh)
-			r.redisLk.Unlock()
+			r.setupLk.Unlock()
```

Comment:
nitpick: When you have multiple places that can err and have to add `Unlock` at multiple places, I find it easier to just create a closure with `defer` - this way if someone adds another statement later it is hard to miss adding the `Unlock`.